core/src/Cache/CacheFallbackOperations.cs

```diff
@@ -238,29 +237,46 @@ internal class CacheFallbackOperations
         }
 
         public static AdalResultWrapper FindMsalEntryForAdal(ITokenCacheAccessor tokenCacheAccessor, string authority,
-            string clientId, string upn)
+            string clientId, string upn, RequestContext requestContext)
```

Comment:
Having so many parameters + static is a code smell - consider refactoring to an instance class with state.

Reply:
I believe this class has been considered a "factory" class or perhaps rather a helper class to create objects... I suppose at this point in time it's even misleading that the class is not marked as static. Clearly some technical debt to be cleaned. IMO, a work item should be created to track refactoring.

---
In reply to: [199527049](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1108#discussion_r199527049) [](ancestors = 199527049)

Reply:
Having too many parameters is bad for maintainability and readability. Many static analyzers have rules for this, including Java's Checkstyle and C#'s FxCop. Making this an instance class would be a way of dealing with this - but there are others o course (an intermediary object, ctor, builder pattern).

Reply:
pending discussion on code style

---
In reply to: [199551935](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1108#discussion_r199551935) [](ancestors = 199551935,199527049)